File: python/euler.py
import math
import time


def euler1():
    return math.exp(1)

def euler2(n: int):
    return (1 + 1 / n) ** n

def euler3(precision: int):
    e = 0
    n = 0
    while True:
        term = 1 / math.factorial(n)
        e += term
        if n > precision:
            break
        n += 1
    return round(e, precision)

if __name__ == '__main__':
    t1 = time.perf_counter_ns()
    r1 = euler1()
    t2 = time.perf_counter_ns()

    t3 = time.perf_counter_ns()
    r2 = euler2(1000000)
    t4 = time.perf_counter_ns()

    t5 = time.perf_counter_ns()
    r3 = euler3(1000)
    t6 = time.perf_counter_ns()

    print(f'euler1: {r1}, {t2 - t1}')
    print(f'euler2: {r2}, {t4 - t3}')
    print(f'euler3: {r3}, {t6 - t5}')

File: go/euler.go
package main

import (
	"math"
	"math/big"
)

func euler1() float64 {
	return math.Exp(1.0)
}

func euler2(n int) float64 {
	return math.Pow(1+1/float64(n), float64(n))
}

func euler3(iterations uint64) float64 {
	var n uint64
	var e float64

	for {
		term := float64(1) / float64(factorial(n))
		e += term
		if n > iterations {
			break
		}
		n += 1
	}
	return e
}

func factorial(n uint64) (result uint64) {

	if n == 0 || n == 1 {
		return 1
	}
	return n * factorial(n-1)
}

func factorial2(n big.Int) (result *big.Int) {
	if n.Cmp(big.NewInt(0)) == 0 || n.Cmp(big.NewInt(1)) == 0 {
		return big.NewInt(1)
	}
	sub := n.Sub(&n, big.NewInt(1))
	return n.Mul(&n, factorial2(*sub))
}
